src/shared/telemetry/telemetryEvent.ts

```diff
@@ -22,47 +22,43 @@ export interface TelemetryEvent {
     data?: Datum[]
 }
 
-export class TelemetryEventArray extends Array<TelemetryEvent> {
-    public toMetricData() {
-        const metricData = new Array<MetricDatum>()
+export function toMetricData(array: TelemetryEvent[]): MetricDatum[] {
+    return ([] as MetricDatum[]).concat(
```

Comment:
This is unnecessary:

```typescript
([] as MetricDatum[]).concat(
    ...array.map(...)
)
```

Just do `return array.map(...)`.

Reply:
Resolved discussion offline